### ISSUE TITLE
Installation test suite failure

### ISSUE THREAD

**Issue by swarder** [firedrakeproject/firedrake#774]
Trying to install Firedrake for the first time, on Ubuntu 14.04 LTS.

After installation, the summary from running the test suite is

`1919 failed, 1544 passed, 52 skipped, 12 xfailed, 35 error in 2380.06 seconds`

Installation script appears successful but contains many warnings; see installation log.

Is it likely that something I've previously installed is messing it up somehow? In which case what is the recommended way for me to achieve a clean installation of Firedrake?

Thanks

[firedrake-install.txt](https://github.com/firedrakeproject/firedrake/files/257180/firedrake-install.txt)

**Comment by miklos1** [comment]
Can you show us how the a test fails? What is the error message of just one failure?

**Comment by swarder** [comment]
I'm not really sure how to interpret the output from running the test suite, but here's a typical section from it, which includes an error:

```
self = Arg(MixedDat((Dat(DataSet(Set((600, 600, 600, 600), 'set_8453'), (2,), 'None_n...41, 441, 441), 'set_8454'), 6, None, 'None_cell_node'))), None, Access('READ'))
update_inc = False

    @collective
    def halo_exchange_begin(self, update_inc=False):
        """Begin halo exchange for the argument if a halo update is required.
            Doing halo exchanges only makes sense for :class:`Dat` objects.
    
            :kwarg update_inc: if True also force halo exchange for :class:`Dat`\s accessed via INC."""
        assert self._is_dat, "Doing halo exchanges only makes sense for Dats"
        assert not self._in_flight, \
>           "Halo exchange already in flight for Arg %s" % self
E       AssertionError: Halo exchange already in flight for Arg OP2 Arg: dat OP2 MixedDat composed of Dats: (Dat(DataSet(Set((600, 600, 600, 600), 'set_8453'), (2,), 'None_nodes_dset'), None, dtype('float64'), 'function_19881[cmpt-0]'), Dat(DataSet(Set((441, 441, 441, 441), 'set_8454'), (1,), 'None_nodes_dset'), None, dtype('float64'), 'function_19881[cmpt-1]')), map OP2 MixedMap composed of Maps: (Map(Set((200, 200, 200, 200), 'plexmesh_cells'), Set((600, 600, 600, 600), 'set_8453'), 3, None, 'None_cell_node'), Map(Set((200, 200, 200, 200), 'plexmesh_cells'), Set((441, 441, 441, 441), 'set_8454'), 6, None, 'None_cell_node')), index None, access OP2 Access: READ

../../local/lib/python2.7/site-packages/pyop2/base.py:457: AssertionError
```

**Comment by miklos1** [comment]
Can you give a full stack trace? Or maybe you could come over to William Penney Lab so that we can check out the problem on your computer?

**Comment by swarder** [comment]
I've now managed to re-install Firedrake on my laptop, and run the test suite with 3498 passed, 52 skipped, 12 xfailed. I'm assuming this is an acceptable result from the test suite?

I don't know how to reproduce the errors I had the first time I installed Firedrake; I'd already tried reinstalling yesterday but with no success. For some reason trying again today for a third time has worked OK.

If it's useful to you for me to bring my laptop to you tomorrow I can do so, but I haven't been able to reproduce the errors.
